MetaMask/metamask-extension
# Bump @metamask/providers to v13.0.0

Driven by a need to get rid of `window.ethereum.networkVersion` as a part of removing networkId work. This PR bumps `@metamask/providers` from v11.1.0 to v13.0.0, which introduces a deprecation warning when accessing `window.ethereum.networkVersion` (as well as `chainId` and `selectedAddress`). The goal is to get this into the wild, alert devs, and to finally remove these properties in the future.

~~Additionally, this PR bumps `browserify` from v16.5.1 to v17.0.0 which is required for polyfilling `streams.pipeline` which is now used in `@metamask/providers`~~ Completed in a separate PR already

https://github.com/MetaMask/providers/compare/v11.1.0...v13.0.0

* Fixes: https://github.com/MetaMask/MetaMask-planning/issues/1068
* See: https://github.com/MetaMask/metamask-extension/pull/20652

## Explanation



## Screenshots/Screencaps



### Before
![Screenshot 2023-09-19 at 9 36 46 AM](https://github.com/MetaMask/metamask-extension/assets/918701/3be060b8-8d76-42ae-b860-2a3d12144326)



### After
![Screenshot 2023-09-19 at 9 31 30 AM](https://github.com/MetaMask/metamask-extension/assets/918701/80676323-c8a6-4d06-bfb4-9b082cee9a64)


## Manual Testing Steps
1. Open browser console in any page
2. Enter `window.ethereum.chainId`
3. You should get the value for the current selected chain ID and a deprecation warning
4. Enter `window.ethereum.chainId` again
5. You should get the value for the current selected chain ID, but no deprecation warning this time
6. Repeat 2-6 for `window.ethereum.networkVersion` and `window.ethereum.selectedAddress`



## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
